Because `ical` no longer depends on `emoji` this breaks mypy for another library:

```
Error: venv/lib/python3.11/site-packages/slixmpp/plugins/xep_0444/stanza.py:9: error: Cannot find implementation or library stub for module named "emoji"  [import-not-found]
venv/lib/python3.11/site-packages/slixmpp/plugins/xep_0444/stanza.py:9: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 6268 source files)
Error: Process completed with exit code 1.
```

It looks like `slixmpp` does not directly depend on emoji, but tries to import it and if it can't be imported it catches the `ImportError` with a fallback
https://github.com/poezio/slixmpp/blob/7a0fb970833c778ed50dcb49c5b7b4043d57b1e5/slixmpp/plugins/xep_0444/stanza.py#L9

After reading the mypy guidance, I'm not sure what the right fix is. I don't think the answer is for `ical` to keep an unnecessary dependency on `emoji`. I'm not sure that what `slixmpp` is doing is wrong. Not sure what mypi is doing is necessarily wrong either or if its a good idea to suppress or how.

Would appreciate feedback on how to proceed.

--- a/homeassistant/components/xmpp/manifest.json
+++ b/homeassistant/components/xmpp/manifest.json
@@ -5,5 +5,5 @@
   "documentation": "https://www.home-assistant.io/integrations/xmpp",
   "iot_class": "cloud_push",
   "loggers": ["pyasn1", "slixmpp"],
-  "requirements": ["slixmpp==1.8.4"]
+  "requirements": ["slixmpp==1.8.4", "emoji==2.8.0"]
 }

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -745,6 +745,9 @@ elkm1-lib==2.2.6
 # homeassistant.components.elmax
 elmax-api==0.0.4
 
+# homeassistant.components.xmpp
+emoji==2.8.0
+
 # homeassistant.components.emulated_roku
 emulated-roku==0.2.1
 

